pkg/catalog/configmeta.go

```diff
@@ -116,6 +136,98 @@ func servicesForRole(
 	return result
 }
 
+// genconfigConnections will look at the cluster spec
+// and generates a map of configmap type and corresponding
+// configmaps to be connected to the given cluster
+func genconfigConnections(
+	cr *kdv1.KubeDirectorCluster,
+) (map[string]map[string]map[string]string, error) {
+
+	models := make(map[string]map[string]string)
+	kdcm := make(map[string]map[string]map[string]string)
+	for _, modelMapName := range cr.Spec.Connections.ConfigMaps {
+		modelCM, err := observer.GetConfigMap(cr.Namespace, modelMapName)
+		if kdConfigMapType, ok := modelCM.Labels["kubedirectorcmtype"]; ok {
+			models[modelMapName] = modelCM.Data
+			kdcm[kdConfigMapType] = models
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return kdcm, nil
+}
+
+// genClusterConnections generates a map of running clusters that are to be connected
+// to this cluster.
+func genClusterConnections(
+	cr *kdv1.KubeDirectorCluster,
+) (map[string]clusterConnections, error) {
+
+	// thisApp, _ := observer.GetApp(cr.Namespace, cr.Spec.AppID)
```

Comment:
Question: Do we want the connectable_to kind of functionality as we did for EPIC? Personally I do not feel the need for it so commented for now

Reply:
Ah I see this is why we had that dangling type declaration before.

I think we do want declarations in the app about what kdclusters and configmaps could/should be connected to. But we should do a fresh design discussion on that (i.e. not copying the EPIC stuff necessarily) and tackle it in some subsequent PR. The feature works without that stuff, it's more of an additional feature.